File: user/src/main/java/com/kjj/user/client/MenuClient.java
package com.kjj.user.client;

import com.kjj.user.dto.menu.MenuNameAndCostDto;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

import java.util.Map;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class MenuClient {

    private final RestTemplate restTemplate;
    private final ClientUriTool uriTool;

    public Boolean existsById(Long menuId) {
        String uri = uriTool.getRequestURI("/v1/client/menu/exist", Map.of("menuId", String.valueOf(menuId)));
        return restTemplate.getForObject(uri, Boolean.class);
    }

    public MenuNameAndCostDto findNameAndCostById(Long menuId) {
        String uri = uriTool.getRequestURI("/v1/client/menu/name-cost", Map.of("menuId", String.valueOf(menuId)));
        return restTemplate.getForObject(uri, MenuNameAndCostDto.class);
    }

    public String findNameById(Long menuId) {
        String uri = uriTool.getRequestURI("/v1/client/menu/name", Map.of("menuId", String.valueOf(menuId)));
        return restTemplate.getForObject(uri, String.class);
    }
}

File: user/src/main/kotlin/com/kjj/user/client/MenuClient.kt
package com.kjj.user.client

import com.kjj.user.dto.menu.MenuNameAndCostDto
import org.springframework.stereotype.Service
import org.springframework.web.client.RestTemplate
import org.springframework.web.client.getForObject
import java.util.Map

@Service
class MenuClient (
    val restTemplate: RestTemplate,
    val uriTool: ClientUriTool
){
    fun existsById(menuId: Long): Boolean {
        val uri = uriTool.getRequestUri(
            "/v1/client/menu/exist",
            mapOf(Pair("menuId", menuId.toString()))
        )
        return restTemplate.getForObject(uri, Boolean::class)
    }

    fun findNameAndCostById(menuId: Long): MenuNameAndCostDto {
        val uri: String = uriTool.getRequestUri(
            "/v1/client/menu/name-cost",
            mapOf(Pair("menuId", menuId.toString())))
        return restTemplate.getForObject(uri, MenuNameAndCostDto::class)!!
    }

    fun findNameById(menuId: Long): String {
        val uri: String = uriTool.getRequestUri(
            "/v1/client/menu/name",
            mapOf(Pair("menuId", menuId.toString()))
        )
        return restTemplate.getForObject(uri, String::class)!!
    }
}